rts/motoko-rts/src/memory/ic/linear_memory.rs

```diff
@@ -6,18 +6,25 @@ use crate::types::*;
 /// Amount of garbage collected so far.
 pub(crate) static mut RECLAIMED: Bytes<u64> = Bytes(0);
 
-// Heap pointer
+// Heap pointer (skewed)
 extern "C" {
-    pub(crate) fn setHP(new_hp: u32);
-    pub(crate) fn getHP() -> u32;
+    fn setHP(new_hp: usize);
```

Comment:
I wonder whether it could make sense to use `Value` for `getHP()` and `setHP()`. Then, it would be more statically checked that one does not mix skewed and unskewed `HP`.

Reply:
Maybe, the functions could be additionally renamed to `setSkewedHeapPointer()` and `getSkewedHeapPointer()` (in the compiler-generated code) because this design is special in comparison to the other unskewed heap boundaries in the RTS.

Reply:
Getting from `Value` to `u32` incurs cost. I am not willing to pay it. Or do you see a way of making it super cheap?

Reply:
Re. `setSkewedHeapPointer`, etc.
`setHP` is now a local function, with only two clients in this file. I am happy with it. (Plus, the motoko side would need a renaming too.) But you wanted to do the _incremental_ adaption, feel free to rename in that PR.

Reply:
> Getting from `Value` to `u32` incurs cost. I am not willing to pay it. Or do you see a way of making it super cheap? Also the arguments passed to `setHP` are more `usize`-like than `Value` like at the callsities.

I thought it is zero cost because `Value` is only a static wrapper. But I may be mistaken.

Reply:
You are right, it is that `from_*` routines that massage stuff. But still, we can change this later when it becomes pressing.